iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -763,6 +764,27 @@ public void alterTableBranchOperation(org.apache.hadoop.hive.ql.metadata.Table h
 
   }
 
+  @Override
+  public void alterTableTagOperation(org.apache.hadoop.hive.ql.metadata.Table hmsTable,
+      AlterTableTagSpec alterTagSpec) {
+    TableDesc tableDesc = Utilities.getTableDesc(hmsTable);
+    Table icebergTable = IcebergTableUtil.getTable(conf, tableDesc.getProperties());
+    Optional.ofNullable(icebergTable.currentSnapshot()).orElseThrow(() ->
+        new UnsupportedOperationException(String.format("Cannot alter tag on iceberg table" +
+            " %s.%s which has no snapshot", hmsTable.getDbName(), hmsTable.getTableName())));
+
+    switch (alterTagSpec.getOperationType()) {
```

Comment:
could we make it reusable and extract common part in `alterTableOperation(hmsTable, alterTag/BranchSpec.getOperationType())`?